### ISSUE TITLE
autoformer predict function needs 4 numpy array for prediction

### ISSUE THREAD

**Issue by pankajr141** [intel-analytics/BigDL#7477]
When going through documentation of autoformer and based on discussion in https://github.com/intel-analytics/BigDL/issues/7456

Training was completed with 
<pre>
forecaster = AutoformerForecaster(past_seq_len=look_back,
                                  future_seq_len=horizon,
                                  input_feature_num=input_feature_num,
                                  output_feature_num=output_feature_num,
                                  label_len=label_len,
                                  freq='h',
                                  seed=2)
forecaster.fit(train_loader, epochs=3, batch_size=32
</pre>

This seems straight forward as fit function require <b>x, y, x_enc, y_enc</b>

what i am unable to understand is why does predict function also need these argument <b>x, y, x_enc, y_enc</b> i.e 4 numpy array (otherwise it just throws error), now as per prediction in general we don't have actual values <b> y </b> available.

can you please provide some information on this.

**Comment by TheaperDeng** [comment]
This a special need for transformer based model (informer, autoformer, fedformer and so on). For y there is a `label_len` that is overlapped with `lookback`, for detailed explanation please see following picture.

<img width="653" alt="image" src="https://user-images.githubusercontent.com/35031544/217712242-f0b76f8c-9c79-4b9a-8381-a0df73652f8e.png">

So, for the case where
```python
look_back = 96
horizon = 720
label_len = 48  # label len is typically set to half of lookback
input_feature_num = 321
output_feature_num = 321
```
The input numpy array for prediction is
```
torch.Size([1, 96, 321]) torch.Size([1, 48, 321]) torch.Size([1, 96, 4]) torch.Size([1, 768, 4])
```
In other word
(batchsize, lookback, input feature num), (batchsize, label len, output feature num), (batchsize, lookback, timeenc dim), (batchsize, label len + horizon, timeenc dim),

More details please see https://arxiv.org/abs/2106.13008

**Comment by pankajr141** [comment]
Thnks for the info

i still have one query, in mentioned dimensions

<pre>
torch.Size([1, 96, 321]) torch.Size([1, 48, 321]) torch.Size([1, 96, 4]) torch.Size([1, 768, 4])
</pre>

(batchsize, lookback, input feature num)
(batchsize, label len, output feature num)
(batchsize, lookback, timeenc dim)
(batchsize, label len + **horizon**, timeenc dim),

Their is still horizon which is exposed in prediction step, which we don't have when performing prediction. How are we generating **(batchsize, label len + horizon, timeenc dim)**

**Comment by TheaperDeng** [comment]
I wonder if this could help you? btw, the 4th input does have horizon, but it is generated only from timestamp, it's reasonable to have future timestamp. @pankajr141

**Comment by pankajr141** [comment]
Yes that helps
